src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h

```diff
@@ -46,8 +46,31 @@ class HumanIK
     Eigen::Matrix<double, 6, 1> m_baseVelocity;
     Eigen::Vector3d m_gravity;
 
+    manif::SO3d I_R_link_manif;
+    manif::SO3Tangentd I_omega_link_manif;
+
+    iDynTree::Rotation I_R_link;
+    iDynTree::AngVelocity I_omega_link;
+
+    struct OrientationTask
+    {
+        /* data */
+        std::shared_ptr<BipedalLocomotion::IK::SO3Task> task;
+        int nodeNumber;
+        iDynTree::Rotation IMU_R_link = iDynTree::Rotation::Identity();
+    };
+
     // tasks
-    std::shared_ptr<BipedalLocomotion::IK::SO3Task> m_link1OrientationTask;
+    OrientationTask m_PelvisTask;
+    OrientationTask m_T8Task;
+    OrientationTask m_RightUpperArmTask;
+    OrientationTask m_RightForeArmTask;
+    OrientationTask m_LeftUpperArmTask;
+    OrientationTask m_LeftForeArmTask;
+    OrientationTask m_RightUpperLegTask;
+    OrientationTask m_RightLowerLegTask;
+    OrientationTask m_LeftUpperLegTask;
+    OrientationTask m_LeftLowerLegTask;
```

Comment:
They can be an std::unorderd_map so you can cycle over them :)

Reply:
Thank you for the advice! I created it here:
https://github.com/ami-iit/biomechanical-analysis-framework/blob/c1d12c4d8edf191e82688d9bca959b97374974cc/src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h#L63-L64